lib/iris/fileformats/pp.py

```diff
@@ -1476,8 +1480,18 @@ def t1(self):
 
         """
         if not hasattr(self, "_t1"):
+            has_year_zero = self.lbyr == 0
+            lbmon, lbdat = self.lbmon, self.lbdat
+            if lbmon == 0 and lbdat == 0:
+                lbmon = lbdat = 1
```

Comment:
It's invalid to have a zero `month` or `day` in any calendar.

However, it is possible to have a PP field with zero values already configured from file loading (not set via property setter methods). 

The approach here is to only ever return a valid `cftime.datetime` regardless. Raising an exception at this point is really not acceptable to users.